allen-cell-animated/simularium-website
# Feature/treat warning messages as warnings

Problem
=======
Now that we get warning, info, and error messages differentiated from the viewer, we should treat warnings and info messages differently from error messages. 

Closes #269 

Solution
========
1. Currently, the `setStatus` action does 2 things: set the viewer status (empty, loading, error, or success) AND set the error message in state. Setting the viewer status to "error" has effects we don't want with a warning or info (for example clearing out the agent panel), and in fact we don't want to change the viewer status when we get a warning or info at all. So I split up `setStatus` into `setStatus` and `setError`. Now, `setStatus` only sets the viewer status (empty, loading, error, or success) and `setError` puts the error message (with various error levels like warning, info, error) in our state so they can be turned into notifications for the user.
2. Modify the `ErrorNotification` component so that it displays warning and info icons (instead of a red X) for warnings and infos

### Other stuff
* Installed esbuild-jest and esbuild (required by esbuild-jest) to modify the jest config, which fixed this annoying problem we've had forever, where the test suite would break if we imported `@aics/simularium-viewer` into src/state/trajectory/logics.ts. This was necessary in this PR because I needed to import simularium-viewer into that logics file! (I'm honestly not sure why this config fixed the problem or why the problem existed in the first place, it was just a lot of trial and error)

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* This change requires updated or new tests

Steps to Verify:
----------------
1. Pull this branch and `npm i` then `npm start`
2. Load a file that needs local geometry files, but don't supply the geometry files
3. You should see some warning messages pop up (instead of error). Everything else should work as before.

Screenshots (optional):
-----------------------
![image](https://user-images.githubusercontent.com/12690133/141230900-07fed054-fc5e-45cc-8b95-5aa338107c1c.png)


Keyfiles (delete if not relevant):
-----------------------
2. src/state/viewer/reducer.ts & src/state/viewer/types.ts - separate out `setStatus` into `setStatus` and `setError`
1. src/components/ErrorNotification/index.tsx - take error level into account when rendering a notification

Almost everything else is just having to update code everywhere `setStatus` has been used.
